### ISSUE TITLE
Make HeaderEditor start early on Firefox

### ISSUE THREAD

**Issue by sebast889** [FirefoxBar/HeaderEditor#120]
Please make a way to make HeaderEditor start early on Firefox so that when Firefox is booted up or pages are restored from session any requests HeaderEditor changes/sends/modifies actually take effect on those pages and they won't be bypassed. 

From reading about this elsewhere webRequest listener has to registered in a certain way for this to happen. More Information on how to start early:
https://bugzilla.mozilla.org/show_bug.cgi?id=1447551
https://bugzilla.mozilla.org/show_bug.cgi?id=1503721

**Comment by sylingd** [comment]
I don't know is webRequest a persistent event or not, but even if it is, HE still require reading rules from database when starting up. It may take a few seconds or longer (depend on how many rules user have). If all requests are blocking in this time, user's browsing experience will be bad.
I think HE can provide some eatra features to make browsing experience better, but it should not make the basic experience worse.

**Comment by sebast889** [comment]
I don't think it will make the browsing experience worse because it will mainly affect the first 1-10 seconds on startup only. Blocking longer than that I think would be rare occurrence. In my opinion the benefit outweighs the negative (where the user will browse thinking HE is working but it isn't).
